### PR TITLE
Pagination forseti

### PR DESCRIPTION
**Pull request by julyto**
Added possibility to get paginate for all connector.
Added paginate for freefloatings

### CODE REVIEW

File: internal/freefloatings/free_floatings_test.go [hunk 1]
```diff
@@ -129,6 +138,41 @@ func TestFreeFloatingsAPIWithDataFromFile(t *testing.T) {
...
+	require.Equal(200, w.Code)
+	err = json.Unmarshal(w.Body.Bytes(), &response)
+	require.Nil(err)
+	require.NotNil(response.FreeFloatings)
+	assert.Len(response.FreeFloatings, 1)
+	// Verify paginate
+	assert.Equal(1, response.Paginate.Start_page)
+	assert.Equal(1, response.Paginate.Items_on_page)
+	assert.Equal(2, response.Paginate.Items_per_page)
+	assert.Equal(3, response.Paginate.Total_result)
```
Comment (kadhikari) [inline internal/freefloatings/free_floatings_test.go:175]: Can you also verify the only free_floating returned by the query here ?

Another test with &distance=100&start_page=1&count=1. Normally there should be an empty free_floating along with a pagination having all 0 s.

File: internal/freefloatings/free_floatings_test.go [hunk 2]
```diff
@@ -129,6 +138,41 @@ func TestFreeFloatingsAPIWithDataFromFile(t *testing.T) {
...
+	require.Equal(200, w.Code)
+	err = json.Unmarshal(w.Body.Bytes(), &response)
+	require.Nil(err)
+	require.NotNil(response.FreeFloatings)
+	assert.Len(response.FreeFloatings, 2)
+	// Verify paginate
+	assert.Equal(0, response.Paginate.Start_page)
+	assert.Equal(2, response.Paginate.Items_on_page)
+	assert.Equal(2, response.Paginate.Items_per_page)
+	assert.Equal(3, response.Paginate.Total_result)
```
Comment (kadhikari) [inline internal/freefloatings/free_floatings_test.go:160]: Can you verify at least one attribute of two elements returned by the query ?

File: internal/freefloatings/context.go [hunk 3]
```diff
@@ -87,11 +88,17 @@ func (d *FreeFloatingsContext) GetFreeFloatings(param *FreeFloatingRequestParame
...
-		if len(resp) > param.Count {
-			resp = resp[:param.Count]
+
+		// Paginate
+		paginate, indexS, indexE := utils.PaginateEndPoint(len(resp), param.Count, param.StartPage)
+		paginate_freefloatings = paginate
+		if indexS >= 0 {
+			resp = resp[indexS:indexE]
+		} else {
+			resp = nil
```
Comment (kadhikari) [inline internal/freefloatings/context.go:98]: We never returned nil before. we initialized to 0 ->  resp := make([]FreeFloating, 0).
Have you tested what the web service response will be if resp = nil here?

**Review by kadhikari — CHANGES_REQUESTED** [review]

File: internal/freefloatings/api.go [hunk 4]
```diff
@@ -121,10 +124,12 @@ func initFreeFloatingRequestParameter(c *gin.Context) (param *FreeFloatingReques
 	var longitude, latitude float64
 	var e error
 	p := FreeFloatingRequestParameter{}
-	countStr := c.DefaultQuery("count", "10")
+	countStr := c.DefaultQuery("count", "25")
 	p.Count = utils.StringToInt(countStr, 10)
```
[the remark this answers is not in the document]
Comment (julyto) [reply internal/freefloatings/api.go:128]: OK

File: internal/freefloatings/free_floatings_test.go (hunk 1 above)
Comment (julyto) [reply internal/freefloatings/free_floatings_test.go:175]: OK

File: internal/freefloatings/free_floatings_test.go (hunk 2 above)
Comment (julyto) [reply internal/freefloatings/free_floatings_test.go:160]: ok

File: internal/freefloatings/context.go (hunk 3 above)
Comment (julyto) [reply internal/freefloatings/context.go:98]: OK

**Review by kadhikari — APPROVED** [review]

**Pull request merged**